client/src/components/layers/BaseLayer.ts

```diff
@@ -0,0 +1,100 @@
+/*eslint class-methods-use-this: "off"*/
+import Track, { Feature, ConfidencePair, TrackId } from '@/lib/track';
+import { FrameDataTrack, CoordinateList } from '@/components/layers/LayerTypes';
+
+import { boundToGeojson } from '@/utils';
+import { StateStyles } from '@/use/useStyling';
+import geo from 'geojs';
+
+
+interface LayerStyle {
+    strokeWidth?: (number | ((a: any, b: any, data: any) => number));
+    strokeOpacity?: (number | ((a: any, b: any, data: any) => number));
+    strokeColor?: (string | ((a: any, b: any, data: any) => string));
+}
+
+interface BaseLayerParams {
+    frameData?: FrameDataTrack;
+    annotator?: any;
+    stateStyling?: StateStyles;
+    typeColorMap?: (t: string) => string;
+    signals?: Record<string, any>;
+}
+
+export default class BaseLayer {
+    frameData?: FrameDataTrack[];
```

Comment:
I don't know that I see the use of keeping this as an instance property.  I see why you need formattedData, but I don't know that you actually need to hold a reference to this value on the class.